catalogue/webapp/components/ConfirmItemRequest/ConfirmItemRequest.tsx

```diff
@@ -184,37 +202,91 @@ const ErrorDialog: FunctionComponent<ErrorDialogProps> = ({ setIsActive }) => (
   </>
 );
 
+type RequestingState = null | 'requesting' | 'confirmed' | 'error';
+
 const ConfirmItemRequest: FunctionComponent<Props> = props => {
   const openButtonRef = useRef<HTMLButtonElement>(null);
-  const { item, work, setIsActive, ...modalProps } = props;
-  const [isConfirmed, setIsConfirmed] = useState(false);
-  const [isLoading, setIsLoading] = useState(false);
-  const [isError, setIsError] = useState(false); // TODO: implement this if something goes wrong with API call
-
+  const { item, work, setIsActive, user, initialHoldNumber, ...modalProps } =
+    props;
+  const [requestingState, setRequestingState] = useState<RequestingState>(null);
+  const [currentHoldNumber, setCurrentHoldNumber] = useState(initialHoldNumber);
   function innerSetIsActive(value: boolean) {
     if (value) {
       setIsActive(true);
-    } else if (isLoading) {
+    } else if (requestingState === 'requesting') {
       // disable close dialog button during api call
     } else {
       setIsActive(false);
-      setIsConfirmed(false);
-      setIsError(false);
     }
   }
 
-  function confirmRequest() {
-    setIsLoading(true);
-    // setIsError(true); // uncomment to test the error UI
+  useEffect(() => {
+    setCurrentHoldNumber(initialHoldNumber);
+  }, [initialHoldNumber]);
+
+  async function confirmRequest() {
+    if (!user) return;
+    setRequestingState('requesting');
+    try {
+      const response = await fetch(
+        withPrefix(`/api/users/${user.userId}/item-requests`),
+        {
+          method: 'POST',
+          body: JSON.stringify({
+            workId: work.id,
+            itemId: item.id,
+            type: 'Item',
+          }),
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        }
+      );
+      if (!response.ok) {
+        setRequestingState('error');
+        // TODO: something to Sentry?
+      } else {
+        setRequestingState('confirmed');
+        setCurrentHoldNumber(currentHoldNumber + 1);
+        // Getting the users current holds, following a successful request, in order to update this number doesn't work because the response we get isn't yet up to date, so changing it manually
```

Comment:
Why doesn't this work?

Reply:
The api response doesn't updated quickly enough, once a request is placed

Reply:
Comment?